Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -170,30 +186,32 @@ void reinitializeJacobians(
         "Inconsistency in global to local transformation during propagation.")
   }
   auto loc = lpResult.value();
-  BoundVector pars;
-  pars << loc[eBoundLoc0], loc[eBoundLoc1], phi(direction), theta(direction),
-      parameters[eFreeQOverP], parameters[eFreeTime];
-  surface.initJacobianToGlobal(geoContext, jacobianLocalToGlobal, position,
-                               direction, pars);
+  // Construct a bound parameters
+  BoundVector boundParams;
+  boundParams << loc[eBoundLoc0], loc[eBoundLoc1], phi(direction),
+      theta(direction), freeParams[eFreeQOverP], freeParams[eFreeTime];
```

Comment:
You might be able to use `transformFreeToBoundParameters` from `EventData/detail/TransformationFreeToBound.hpp` here instead of doing it manually.

Reply:
Exactly. Changed now.